### PR TITLE
Tool 574 support multiple identities in p12

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: certificateutil/info_model.go [hunk 1]
```diff
@@ -22,32 +22,21 @@ type CertificateInfoModel struct {
...
-	printable["expire"] = info.EndDate.String()
-
-	errors := []string{}
-	if err := info.CheckValidity(); err != nil {
-		errors = append(errors, err.Error())
-	}
-	if len(errors) > 0 {
-		printable["errors"] = errors
-	}
+	team := fmt.Sprintf("%s (%s)", info.TeamName, info.TeamID)
```
Comment (lpusok) [inline certificateutil/info_model.go]: Check Cert and profile installer step logging

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+[[constraint]]
+  branch = "pkcs12"
```
Comment (godrei) [inline Gopkg.toml]: do not forget to change back to master

File: certificateutil/filter.go [hunk 3]
```diff
@@ -13,22 +13,40 @@ func FilterCertificateInfoModelsByFilterFunc(certificates []CertificateInfoModel
...
-				certificateInfosByName[certificateInfo.CommonName] = certificateInfo
-			}
+		if certificateInfo.CheckValidity() != nil {
+			invalidCertificates = append(invalidCertificates, certificateInfo)
+			continue
+		}
+		activeCertificate, ok := certificateInfosByName[certificateInfo.CommonName]
+		if !ok {
+			certificateInfosByName[certificateInfo.CommonName] = certificateInfo
+		} else if certificateInfo.EndDate.After(activeCertificate.EndDate) {
```
Comment (godrei) [inline certificateutil/filter.go]: what if a cert with common name A expires 2016.01.02, found first, then
cert with common name A expires 2016.01.01 <- this will not be added to the `duplicatedCertificates`, since the expire is before the first cert's expire

File: certificateutil/info_model.go [hunk 4]
```diff
@@ -112,7 +103,15 @@ func InstalledCodesigningCertificateInfos() ([]CertificateInfoModel, error) {
 	if err != nil {
 		return nil, err
 	}
-	return CertificateInfos(certificates), nil
+
+	infos := []CertificateInfoModel{}
```
Comment (godrei) [inline certificateutil/info_model.go]: why was this change needed?

File: certificateutil/info_model.go [hunk 5]
```diff
@@ -80,30 +74,27 @@ func NewCertificateInfo(certificate x509.Certificate) CertificateInfoModel {
...
 // NewCertificateInfosFromPKCS12 ...
 func NewCertificateInfosFromPKCS12(pkcs12Pth, password string) ([]CertificateInfoModel, error) {
-	certificates, err := CertificatesFromPKCS12File(pkcs12Pth, password)
+	identities, err := CertificatesFromPKCS12File(pkcs12Pth, password)
 	if err != nil {
 		return nil, err
 	}
-	return CertificateInfos(certificates), nil
+
+	infos := []CertificateInfoModel{}
```
Comment (godrei) [inline certificateutil/info_model.go]: InstalledCodesigningCertificates

File: certificateutil/info_model.go (hunk 1 above)
Comment (lpusok) [reply certificateutil/info_model.go]: certificate and profile installer does not use this function.

File: certificateutil/info_model.go (hunk 5 above)
Comment (lpusok) [reply certificateutil/info_model.go]: removed NewCertificateInfosFromPKCS12 function

File: certificateutil/filter.go (hunk 3 above)
Comment (lpusok) [reply certificateutil/filter.go]: fixed

File: certificateutil/filter.go [hunk 6]
```diff
@@ -22,26 +22,36 @@ type ValidCertificateInfo struct {
...
-	validCertificates := []CertificateInfoModel{}
-	for _, validCertificate := range certificateInfosByName {
-		validCertificates = append(validCertificates, validCertificate)
+	var validCertificates, duplicatedCertificates []CertificateInfoModel
+	for _, certs := range nameToCerts {
+		if len(certs) == 0 {
+			continue
+		}
+
+		latestCert := certs[0]
```
Comment (godrei) [inline certificateutil/filter.go]: NOTE: what about this

```
		sort.Slice(certs, func(i, j int) bool {
			return certs[i].EndDate.Before(certs[j].EndDate)
		})

		validCertificates = append(validCertificates, certs[0])
		duplicatedCertificates = append(duplicatedCertificates, certs[1:]...)
```

is not it a bit cleaner?

Comment (lpusok) [reply certificateutil/filter.go]: It is more clear, even if less optimal in theory.

File: Gopkg.toml (hunk 2 above)
Comment (lpusok) [reply Gopkg.toml]: fixed
